File: IT/Programmazione funzionale/Esercizi/C++/esercizio27.cpp
/* file esercizio27.cpp */

#include <iostream>

using std::cout;
using std::endl;

void forn(int n) {
	if (n > 0) {
		forn(n-1);
		cout << n <<  " ";
	}
}

int main() {
	forn(10);
	cout << endl;
}

File: IT/2/Programmazione funzionale/Esercizi/Java/Esercizio27.java
/* file Esercizio27.java */

public class Esercizio27 {
	private static void forn(int n) {
		if (n > 0) {
			forn(n-1);
			System.out.print(n + " ");
		}
	}

	public static void main(String [] args) {
		forn(10);
		System.out.print(System.lineSeparator());
	}
}

